insights/parsers/logrotate_conf.py

```diff
@@ -0,0 +1,163 @@
+"""
+LogrotateConf - files ``/etc/logrotate.conf`` and others
+========================================================
+
+Classes to parse logrotate confuration files:
+- ``/etc/logrotate.conf``
+- ``/etc/logrotate.d/*``
+
+See: http://www.linuxmanpages.org/8/logrotate
+
+"""
+
+
+from .. import parser, Parser, get_active_lines, LegacyItemAccess
+from insights.specs import Specs
+
+PAIRED_OPTS = (
+        'prerotate',
+        'postrotate',
+        'firstaction',
+        'lastaction',
+)
+
+
+@parser(Specs.logrotate_conf)
+class LogrotateConf(Parser, LegacyItemAccess):
+    """
+    Class for parsing ``/etc/logrotate.conf`` and ``/etc/logrotate.d/*``
+    configuration files.
+
+    Sample logrotate configuration file::
+
+        # sample file
+        compress
+
+        /var/log/messages {
+            rotate 5
+            weekly
+            postrotate
+                        /sbin/killall -HUP syslogd
+            endscript
+        }
+
+        "/var/log/httpd/access.log" /var/log/httpd/error.log {
+            rotate 5
+            mail www@my.org
+            size=100k
+            sharedscripts
+            postrotate
+                        /sbin/killall -HUP httpd
+            endscript
+        }
+
+        /var/log/news/news.crit
+        /var/log/news/olds.crit  {
+            monthly
+            rotate 2
+            olddir /var/log/news/old
+            missingok
+            postrotate
+                        kill -HUP `cat /var/run/inn.pid`
+            endscript
+            nocompress
+        }
+
+    Examples:
+        >>> type(log_rt)
+        <class 'insights.parsers.logrotate_conf.LogrotateConf'>
+        >>> log_rt.options
+        ['compress']
+        >>> log_rt.log_files
+        ['/var/log/messages', '/var/log/httpd/access.log', '/var/log/httpd/error.log', '/var/log/news/news.crit', '/var/log/news/olds.crit']
+        >>> log_rt['compress']
+        True
+        >>> 'weekly' in log_rt['/var/log/messages']
+        True
+        >>> log_rt['/var/log/messages']['postrotate']
+        ['/sbin/killall -HUP syslogd']
+        >>> log_rt.options_of_logfile('/var/log/httpd/access.log')['mail']
+        'www@my.org'
+
+    Attributes:
+        data(dict): All parsed options and log files are stored in this
+            dictionary.
+    """
+
+    def parse_content(self, content):
+
+        def _parse_opts(line):
+            if '=' in line:
+                l_sp = line.split('=')
+            else:
+                l_sp = line.split(None, 1)
+            return l_sp[0], l_sp[1] if len(l_sp) == 2 else True
```

Comment:
@xiangce I believe you'll get an exception if you return `True` since the calling code expects a tuple.

Reply:
@bfahr - Yep, I think I did return a tuple with two elements:
- The first element is `l_sp[0]`
- The second one is `l_sp[1]`, but when there is only one element in `l_sp`, `True` will be the second element.
   `True` is for the options without a value, such as `compress`

Reply:
Thanks for the clarification @xiangce, that looks correct.